src/plugins/launch/slurm/launch_slurm.c

```diff
@@ -590,6 +590,53 @@ extern int launch_p_create_job_step(srun_job_t *job, bool use_all_cpus,
 	return SLURM_SUCCESS;
 }
 
+static char *_find_quote_token(char *tmp, char *sep, char **last)
+{
+	char *start;
+	int i, quote_single = 0, quote_double = 0;
+
+	xassert(last);
+	if (*last)
+		start = *last;
+	else
+		start = tmp;
+	if (start[0] == '\0')
+		return NULL;
+	for (i = 0; ; i++) {
```

Comment:
Yeah, we're not going to do this loop without an end condition, even if you think it is not needed.